enhancements/network/host-port-registry.md

```diff
@@ -0,0 +1,111 @@
+---
+title: Host Port Registry
+authors:
+  - "@squeed"
+reviewers:
+  - "@dcbw"
+  - "@jsafrane"
+  - "@
+approvers:
+  - "@smarterclayton"
+creation-date: 2020-08-26 
+last-updated: 2020-08-26
+status: implementable
+---
+
+# Host Port Registry
+
+OpenShift has a number of host-level services that listen on a socket. Since they
+are not isolated by network namespace, they can conflict if both listen on the same
+port.
+
+This document serves as the authoritative list of port assignments.
+
+## Background
+
+Ports 9000-9999 are available for general use by system services. We require that
+customers open these ports between nodes in a cluster.
+
+These ports are used by certain services that, for whatever reason, must run
+in the host network (or use host-port forwarding).
+
+The Kubernetes scheduler is aware of host ports, and will fail to schedule pods
+on a node where there would be a port conflict.
+
+### Localhost
+
+Separately, it is a common pattern for a pod to have one or more processes only
+listening on localhost. Since all HostNetwork pods share a network namespace,
+coordination is also required in this case.
+
+## Requirements
+
+All processes that wish to listen on a host port MUST
+
+- Have an entry in the table below
+- be in a (user documented range)[https://github.com/openshift/openshift-docs/blob/master/modules/installation-network-user-infra.adoc], 
+  if they are intended to be reachable 
+- declare that port in their `Pod.Spec`
+
+Localhost-only ports SHOULD be outside of this range, to leave room.
+
+# Port Registry
+
+## External
+
+All ports are TCP.
+
+| Port  | Process   | Owner | Since | Notes |
+|-------|-----------|-------|-------|-------|
+| 2379  | etcd      | etcd || |
+| 2380  | etcd      | etcd || |
+| 6443  | kube-apiserver | apiserver || |
+| 9001  | machine-config-daemon oauth proxy | node || metrics |
+| 9100  | node-exporter | monitoring |||
+| 9101  | openshift-sdn kube-rbac-proxy | network || metrics|
+| 9102  | ovn-kubernetes kube-rbac-proxy | network || metrics|
+| 9103  | ovn-kubernetes kube-rbac-proxy | network || metrics|
+| 9537  | crio      | node || metrics |
+| 9641  | ovn-kubernetes northd | network | 4.3 ||
+| 9642  | ovn-kubernetes southd | network | 4.3 ||
+| 9643  | ovn-kubernetes northd | network | 4.3 ||
+| 9644  | ovn-kubernetes southd | network | 4.3 ||
+| 9978  | etcd      | etcd || metrics |
+| 9979  | etcd      | etcd || ? |
+| 10010 | crio | node || stream port|
+| 10250 | kubelet | node || kubelet api |
+| 10251 | kube-scheduler | apiserver || healthz |
+| 10256 | openshift-sdn | network || healthz |
+| 10257 | kube-controller-manager | apiserver || metrics + healthz |
+| 10259 | kube-scheduler | apiserver || metrics |
+| 10357 | cluster-policy-controller | apiserver || healthz |
+| 17697 | kube-apiserver | apiserver || ? |
+| 22623 | machine-config-server | node |||
+| 22624 | machine-config-server | node |||
+| 29101 | openshift-sdn | network || metrics |
+
+
+## Localhost-only
```

Comment:
For the on-prem platforms (IPI baremetal, openstack, vsphere, ovirt) we  have also on the master nodes the ports used for the self-hosted API LB. 

9445 is the LB frontend and 50000 used for LB statistics

https://github.com/openshift/baremetal-runtimecfg/blob/master/cmd/monitor/monitor.go#L55-#L56